auth/auth_test.go

```diff
@@ -0,0 +1,260 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/Gimulator/Gimulator/config"
+	"github.com/Gimulator/Gimulator/object"
+)
+
+type TestInput struct {
+	ConfigPath string
+	Pass       bool
+	id         string
+	hash       string
+}
+
+var cp = []string{
+	"example/roles.yaml",
+	"example/roles1.yaml",
+	"example/roles2.yaml",
+	"example/roles3.yaml",
+	"example/roles4.yaml",
+	"example/",
+	"WrongPath1",
+	"WrongPath2",
+}
+
+func GetResult(res bool, e error) (string, bool) {
+	if e != nil && res == false {
+		return fmt.Sprintf("Test has error and expected to be failed: %s", e), true //true negative
+	} else if res == false && e == nil {
+		return "Test passed but expected to have error.", false //false positive
+	} else if res == true && e != nil {
+		return fmt.Sprintf("Test has error but expected to be passed. error: %s", e), false //false negative
+	}
+	return "Test has no error and passed", true //true positive
+}
+func PrintResult(t *testing.T, res bool, e error) {
+	s, b := GetResult(res, e)
+	if b != true {
+		t.Errorf("FAILED UNEXPECTEDLY: %s", s)
+	} else {
+		t.Logf("PASSED: %s", s)
+	}
+}
+func TestNewAuth(t *testing.T) {
+	testCases := []TestInput{
+		{ConfigPath: cp[0], Pass: true},
+		{ConfigPath: cp[1], Pass: true},
+		{ConfigPath: cp[2], Pass: true},
+		{ConfigPath: cp[5], Pass: false},
+		{ConfigPath: cp[6], Pass: false},
+		{ConfigPath: cp[7], Pass: false},
+	}
+	for tc, _ := range testCases {
```

Comment:
I cannot understand this part.
Why did you use nested loop? and for both of them declared tc variable.
A better approach is:
for _, test := range testCases {
    statements
}

Reply:
Fixed